### DISCUSSION TITLE
Need some advice on TUN_BUILDER on windows

### DISCUSSION THREAD

**Discussion by BaldMonkey77** [OpenVPN/openvpn3#325] [Q&A] [answered]
ATM, I believe that the tun builder package is not designed to work on Windows. (user of unistd.h and posix:: in asio). Am I reading  the code correctly ? If my assumptions are right, I was wondering about what the path forward for tun builder in windows is ?

**Comment by lstipakov** [comment]
You are right, at the moment tunbuilder doesn't work on Windows. Regarding the path forward - it might help if you tell us what do you want to achieve.

**Reply by BaldMonkey77** [reply]
Firstly, thanks for promptly responding.  To answer your question, I am trying to create a client that manages multiple OpenVPN connections to a bunch of servers, while not directly exposing an interface on the system. The routing and packet generation is done by an user-space process. The plan is to use a pipe-like (depending on platform) to hook up the connections to the user-space process.

**Reply by schwabe** [reply, marked as answer]
Then tun builder interface actually sounds like a good idea for that task. The dpendencies on Unix might be accidential since I don't see a good reason that we need them, so fixing/refactoring it a little bit to get rid of these dependencies is probably okay. Patches are welcome as always :)
